scripts/check_branch.py

```diff
@@ -71,15 +102,30 @@ def main(argv):
                         help='Print the deployment matching the current branch or exit '
                              'with non-zero status code if no such deployment exists.')
     args = parser.parse_args(argv)
-    branch = gitlab_branch() or local_branch()
+    branch_in_context = branch_context()
+    if branch_in_context.platform == 'github' and branch_in_context.base_ref:
+        branch = branch_in_context.base_ref
+    else:
+        branch = branch_in_context.name
+    deployment = default_deployment(branch)
     if args.print:
-        deployment = default_deployment(branch)
         if deployment is None:
             sys.exit(1)
         else:
             print(deployment)
     else:
         check_branch(branch, config.deployment_stage)
+        allowed_deployments = config.shared_deployments_for_branch(branch)
+        if deployment == 'prod':
```

Comment:
These checks seem at least partially redundant with the call to `check_branch` on line #117.  If `deployment == 'prod'` and `branch != 'prod'`, won't we have already raised an exception? I suspect there's a clearer way to express this logic.

Reply:
It is redundant, as you point out. I may have been overthinking the solution.